### ISSUE TITLE
Globally installed console can't bootstrap

### ISSUE THREAD

**Issue by cburschka** [hechoendrupal/drupal-console#3195]
To reproduce:

- Install in ~/.composer using `composer global require drupal/console`.
- Go into a Drupal directory that is fully installed, but doesn't have the drupal/console package in its vendor folder.
- Try to run a console command.

      Fatal error: Uncaught InvalidArgumentException: 
      The file ".../vendor/drupal/console-core//services.yml" does not exist. 
      in /home/cburschka/.composer/vendor/symfony/config/FileLocator.php on line 44

The error message indicates that the services.yml file is looked for in the local vendor path, even though the console itself is installed in ~/.composer/vendor/.

(I'm not sure if or how this can be fixed in drupal console, but it'd be nice to be able to maintain a single console install, especially when working with many different Drupal codebases.)

**Comment by dagomar** [comment]
I have the exact same issue right now. How to solve...

**Comment by cburschka** [comment]
Update: Even installing the same version of drupal/console locally won't allow the global command to work. You need to explicitly call the local script with `./vendor/bin/drupal`.

The relevant error is

```
  [Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException]  
  You have requested a non-existent service "class_loader".                   
```

**Comment by jmolivas** [comment]
DrupalConsole must be installed per-site we still provide a global executable.

More information  at the docs => https://docs.drupalconsole.com/en/getting/project.html

Closing this one. Feel free to comment if still happening after following proposed instructions.

**Comment by xmacinfo** [comment]
Nevertheless, this is still a pain to install. Lot's of developers abandon Drupal Console instead.

Any way to install Both Console without Composer?

**Comment by jmolivas** [comment]
@xmacinfo why and how are these instructions painful or hard to follow?
More information about getting the project => https://docs.drupalconsole.com/en/getting/project.html

### Install per site 
More info => https://docs.drupalconsole.com/en/getting/composer.html

Change directory to Drupal site:
```
cd /path/to/drupal8.dev
```
Execute composer require command:
```
composer require drupal/console:~1.0 \
--prefer-dist \
--optimize-autoloader
```
Download Drupal + DrupalConsole using DrupalComposer project template
```
composer create-project \
drupal-composer/drupal-project:8.x-dev \
drupal8.dev \
--prefer-dist \
--no-progress \
--no-interaction
```
Update DrupalConsole
```
composer update drupal/console --with-dependencies
```

### Install Launcher aka global executable
More info => https://docs.drupalconsole.com/en/getting/launcher.html
```
curl https://drupalconsole.com/installer -L -o drupal.phar
mv drupal.phar /usr/local/bin/drupal
chmod +x /usr/local/bin/drupal
```

**Comment by xmacinfo** [comment]
Greetings!

Instructions are easy to follow and for most Drupal 8 sites, I can successfully run Drupal Console (the global installation + the site installation). Sadly, there are always a few Drupal 8 installation where Drupal Console will not launch, no matter the steps I do to try to make it work.

For those reluctant sites, I do not know yet if I will delete the vendor folder or move the content and configuration to another install.

**Comment by jmolivas** [comment]
@xmacinfo when composer fails it show you which dependencies are causing the issues I highly recommend you to read this blog post [Using Composer to debug issues updating Drupal core](https://www.previousnext.com.au/blog/using-composer-debug-issues-updating-drupal-core)
